back/routes/adventure.js

```diff
@@ -24,4 +24,106 @@ router.post('/restaurants', async(req, res) => {
+Parse.serverURL = 'http://parseapi.back4app.com';
+
+// ----- Get current user's preferences -----
+router.post('/preferences/current', async(req, res) => {
+  const query = new Parse.Query('UserPreference');
+  query.equalTo("username", req.body.username)
+  try {
+    let currentPreferences = await query.find();
+    res.status(200);
+    res.send(currentPreferences[0].toJSON().activePreferences);
+  } catch(error) {
+    res.status(400);
+    res.send(error);
+  }
+})
+
+// ----- Get all existing Preferences -----
+router.get('/preferences/all', async(req, res) => {
+  const query = new Parse.Query('Preference');
+  try {
+    let preferences = await query.find();
+    res.status(200);
+    res.send(preferences);
+  } catch(error) {
+    res.status(400);
+    res.send(error);
+  }
+})
+
+// ----- Get all inactive preferences -----
+router.post('/preferences/inactive', async(req, res) => {
+  // req.body elements
+  const username = req.body.username
+  // Find current preferences
+  const getUserPreferenceQuery = new Parse.Query('UserPreference');
+  getUserPreferenceQuery.equalTo("username", username)
+  let activePreferences = null;
+  try {
+    let userPreferences = await getUserPreferenceQuery.find();
+    activePreferences = userPreferences[0].toJSON().activePreferences;
+  } catch(error) {
+    res.status(400);
+    res.send(error);
+  }
+  // Get all possible preferences
+  const allPreferencesQuery = new Parse.Query('Preference');
+  let allPreferences = null;
+  try {
+    allPreferences = await allPreferencesQuery.find();
+  } catch(error) {
+    res.status(400);
+    res.send(error);
+  }
+  // Find inactive preferences
+  let inactivePreferences = allPreferences.filter((preference) => !activePreferences.includes(preference.toJSON().objectId));
+  try {
+    res.status(201);
+    res.send(inactivePreferences);
+  } catch(error) {
+    res.status(400);
+    res.send(error);
+  }
+})
+
+// ----- Update user's preferences ------
+router.post('/preferences/update', async(req, res) => {
+  // req.body elements
+  const prioritize = req.body.prioritize;
+  const activePreferences = req.body.activePreferences;
+  const username = req.body.username;
+  // Define changing elements
+  let differentPriority = false;
+  let differentActivePrefences = false;
+  if (req.body.prioritize !== null) differentPriority = true;
+  if (req.body.activePreferences !== null) differentActivePrefences = true;
+  // Find objectId of the current user preference
+  const findQuery = new Parse.Query('UserPreference');
+  findQuery.equalTo("username", username)
+  let objectId = null;
+  try {
+    let currentPreferences = await findQuery.find();
+    objectId = currentPreferences[0].toJSON().objectId
+  } catch(error) {
+    res.status(401);
+    res.send(error);
+  }
+  // Update information for that user
+  let updateQuery = new Parse.Object('UserPreference');
+  updateQuery.set('objectId', objectId)
```

Comment:
I'm surprised there isn't a simpler way to do this. We can circle back to this to see if we can clean this up.